balena/models/device.py

```diff
@@ -1766,7 +1776,7 @@ def start_os_update(self, uuid_or_id: Union[str, int], target_os_version: str) -
         data = {"parameters": {"target_version": target_os_version}}
 
         url_base = self.__config.get_all()["deviceUrlsBase"]
-        action_api_version = self.__settings.get("device_actions_endpoint_version")
+        action_api_version = "v2" if run_detached is True else self.__settings.get("device_actions_endpoint_version")
```

Comment:
```suggestion
action_api_version = "v2" if run_detached else self.__settings.get("device_actions_endpoint_version")
```

Reply:
v2 is only for detached HUP at this point @otaviojacobi

Reply:
Good catch on the "is True", @otaviojacobi . If the user happens to use runDetached="True", that still works with your change. I'm a little surprised use of a string isn't type checked though since it's typed as `Optional[bool]` .

Reply:
Yes, the key thing here is to (similar to typescript) always think of the type system as best effort. There is only so much you can do on a dynamically typed languages (js/python) to avoid these mistakes. There is the argument of if someone does `run_detatched="False"` it will still be truthy, however considering it defaults to `False` I think someone messing with this parameter probably wants to run detatched

Reply:
We can also always validate that it is either  `True` `False` or `None` and throw it otherwise so that we avoid all these sort of issues. I would defer if you want to go this route to you

Reply:
Makes sense. Added below.

```
        if not isinstance(run_detached, bool) and run_detached is not None:
            raise ValueError(f"run_detached must be True, False, or None, got {type(run_detached)}: {run_detached}")
        action_api_version = "v2" if run_detached is True else self.__settings.get("device_actions_endpoint_version")
```

Reply:
I don't see why we need to allow the value to be None. We have  a default value of False. How about:

```
        if not isinstance(run_detached, bool):
            raise ValueError(f"run_detached must be True or False, got {type(run_detached)}: {run_detached}")
        action_api_version = "v2" if run_detached else self.__settings.get("device_actions_endpoint_version")
```